### PR TITLE
Added resource matchup suggestion for compare-factory-config.py (SOFTWARE-4327)

### PR DESCRIPTION
**Pull request by Caffretro**
## 10/24/2020
- Added resource matchup suggestion function for factory entries
- Formatted topology tree dumping
- Formatted output
- Fixed bug that ignores factory entries in yml files

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ src/webapp/*.pyc
 src/config.py
 */.DS_Store
 .vscode
+bin/country-names-unify.py
```
Comment (brianhlin) [inline .gitignore:9]: I think you missed a commit to remove

File: bin/compare-factory-config.py [hunk 2]
```diff
@@ -176,27 +192,33 @@ def run(argv):
...
-    nonmatch_all_entries = find_non_topology_matches(gfactory_DB, topology_DB)
+    nonmatch_all_entries = find_non_topology_matches(
+        gfactory_DB, topology_DB, resources)
 
     # output formatted results
+    print(f'\nOutput format: <GLIDEIN_ResourceName>:<corresponding factory entry name>:<suggestion for a resource match>\n')
     print(f'\nFactory entries that match a Topology entity other than a resource: \n')
     for x in match_nonresource_entries:
-        print(f'- {x[0]}: {x[1]}')
+        print(f'{x[0]}:{x[1]}:{x[2]}')
```
Comment (brianhlin) [inline bin/compare-factory-config.py:218]:
```suggestion
        print(f'{x[0]},{x[1]},{x[2]}')
```

May as well make the output CSV-like if we're going to be dumping things into a spreadsheet!

File: bin/compare-factory-config.py [hunk 3]
```diff
@@ -176,27 +192,33 @@ def run(argv):
...
     print(f'\nFactory entries that do not match any entity in Topology: \n')
-    for x in sorted(set(nonmatch_all_entries)):
-        print(f'- {x}')
+    for x in nonmatch_all_entries:
+        print(f'{x[0]}:{x[1]}:{x[2]}')
```
Comment (brianhlin) [inline bin/compare-factory-config.py:221]:
```suggestion
        print(f'{x[0]},{x[1]},{x[2]}')
```

File: bin/compare-factory-config.py [hunk 4]
```diff
@@ -176,27 +192,33 @@ def run(argv):
...
     match_nonresource_entries = find_non_resource_matches(
-        gfactory_DB, topology_DB)
+        gfactory_DB, topology_DB, resources)
     # Entry names corresponding to GLIDEIN_ResourceNames above
-    nonmatch_all_entries = find_non_topology_matches(gfactory_DB, topology_DB)
+    nonmatch_all_entries = find_non_topology_matches(
+        gfactory_DB, topology_DB, resources)
 
     # output formatted results
+    print(f'\nOutput format: <GLIDEIN_ResourceName>:<corresponding factory entry name>:<suggestion for a resource match>\n')
```
Comment (brianhlin) [inline bin/compare-factory-config.py:215]: Is this the actual format? If so, could you switch the first and second columns?

**Review by brianhlin — APPROVED** [review]
Some minor nitpicks but otherwise LGTM

File: bin/compare-factory-config.py (hunk 2 above)
Comment (Caffretro) [reply bin/compare-factory-config.py:218]: Oh I was using columns for delimiters and parsing them locally, but yeah this is better

File: bin/compare-factory-config.py (hunk 4 above)
Comment (Caffretro) [reply bin/compare-factory-config.py:215]: that's a mistake here, output was correct and description is wrong

**Comment by brianhlin** [comment]
Thanks for the changes! Could you populate the spreadsheet with the output as soon when you get the chance?

**Pull request merged**